README.md

```diff
@@ -2,14 +2,146 @@
+
+Tuxedo makes the process of developing UTXO-based runtimes faster and safer by freeing developers from having to re-implement all of the common and error-prone UTXO logic in each chain. It also makes the process more standard by providing developers with simple interfaces for their Tuxedo Pieces. When developing a Tuxedo piece, a developer will complete some or all of these following tasks.
+
+### Declaring Data Types
+
+If the Tuxedo piece has any custom data types, they must be declared by implementing the [`UtxoData` trait](https://off-narrative-labs.github.io/Tuxedo/tuxedo_core/dynamic_typing/trait.UtxoData.html). For example, a crytpocurrency may have a data type called `Coin` or a voting solution may have data types called `Poll` and `Vote`. The developer only has to declare the data type, no notion of a "storage item" because there is no global state in the UTXO model. All state is local to individual UTXOs.
+
+### Defining Transaction Constraints
+
+All Tuxedo pieces will define one or more sets of constraints that a transaction must satisfy to be valid. This is done through a [dedicated trait](https://off-narrative-labs.github.io/Tuxedo/tuxedo_core/trait.Verifier.html) (whose name will be changed very soon and is therefore not mentioned in the Readme). Unlike the accounts model, the Tuxedo piece is not responsible for calculating the final state after the transaction. Rather the final state is passed in as the transaction's output set. The piece only verifies that the appropriate constraints are met. For example it may check that in an on-chain chess game, the input piece really is allowed to move to the location specified in the output. In a more classic example, it will check tha the tokens that are being spent are of greater or equal value to the tokens being created.
+
+### Declaring Redemption Logic
+
+Each individual UTXO in the UXO set is protected by a piece of associated logic that determines when it may be spent. This lgic is defined in the [`Redeemer` Trait](https://off-narrative-labs.github.io/Tuxedo/tuxedo_core/redeemer/trait.Redeemer.html). The most classic example is that the UTXO is owned by a particular public key and the transaction must be signed by that key in order to unlock the input. Other examples also exist. For example, the UTXO may be claimable by anyone, or by nobody at all, or it may require a valid proof or work to be consumed.
```

Comment:
Typo

Reply:
```suggestion
Each individual UTXO in the UTXO set is protected by a piece of associated logic that determines when it may be spent. This logic is defined in the [`Redeemer` Trait](https://off-narrative-labs.github.io/Tuxedo/tuxedo_core/redeemer/trait.Redeemer.html). The most classic example is that the UTXO is owned by a particular public key and the transaction must be signed by that key in order to unlock the input. Other examples also exist, such as, the UTXO may be claimable by anyone, or by nobody at all, or it may require a valid proof or work to be consumed.
```